clients/banking/src/utils/validations.ts

```diff
@@ -107,6 +121,22 @@ export const validateTodayOrAfter: Validator<string> = value => {
   }
 };
 
+export const validateDateWithinNextYear: Validator<string> = value => {
```

Comment:
Instead of having the validator returning `void` when the value is empty, maybe it could be better to create one util:

```ts
const optional =
  <T extends string>(validator: Validator<T>) =>
  (value: T): ValidatorResult => {
    const result = validator(value);

    if (result != null && value !== "") {
      if (typeof result === "string") {
        return result;
      }

      return result.then(asyncResult => {
        if (asyncResult != null && value !== "") {
          return asyncResult;
        }
      });
    }
  };
```

Usage:

`validate: optional(validateTransferReference)`